app/src/main/java/unipd/se18/ocrcamera/AdapterTestElement.java

```diff
@@ -64,79 +66,101 @@ public long getItemId(int position) {
     }
 
     @Override
-    public View getView(int position, View convertView, ViewGroup parent) {
+    public View getView(final int position, View convertView, ViewGroup parent) {
         if (convertView == null) {
             convertView = LayoutInflater.from(context).inflate(R.layout.test_element, parent, false);
         }
 
         // Set the correctness value
-        try {
-            TextView correctness = convertView.findViewById(R.id.correctness_view);
-            float confidence = entries[position].getConfidence();
-            String confidenceText = new DecimalFormat("#0").format(confidence) + " %";
-
-            // Set the color of the correctness
-            if(confidence < 70) {
-                correctness.setTextColor(Color.RED);
-            } else if (confidence < 85) {
-                correctness.setTextColor(Color.YELLOW);
-            } else {
-                correctness.setTextColor(Color.GREEN);
-            }
 
-            correctness.setText(confidenceText);
-
-            // Set the name of the pic
-            TextView name = convertView.findViewById(R.id.pic_name_view);
-            String picName = entries[position].getFileName();
-            name.setText(picName);
-
-            // Set the pic view
-            ImageView analyzedPic = convertView.findViewById(R.id.pic_view);
-            Bitmap img = entries[position].getPicture();
-
-            // Scaling the pic view
-            int imgWidth = img.getWidth();
-            int imgHeight = img.getHeight();
-            WindowManager mWindowManager = (WindowManager) convertView.getContext()
-                    .getSystemService(Context.WINDOW_SERVICE);
-            DisplayMetrics mDisplayMetrics = new DisplayMetrics();
-            Display mDisplay = mWindowManager.getDefaultDisplay();
-            mDisplay.getMetrics(mDisplayMetrics);
-            int scaledWidth = mDisplayMetrics.widthPixels;
-            int scaledHeight = (scaledWidth*imgHeight)/imgWidth;
-
-            Log.v(TAG,"pic \"" + picName + "\" scaled from " + imgWidth + "x" + imgHeight +
-                    " to " + scaledWidth + "x" + scaledHeight);
-            analyzedPic.setImageBitmap(Bitmap.createScaledBitmap(img, scaledWidth, scaledHeight,false));
-
-            // Set the Tags text
-            TextView tags = convertView.findViewById(R.id.tags_view);
-            StringBuilder assignedTags = new StringBuilder();
-            for(String tag: entries[position].getTags()) {
-                assignedTags.append(tag).append(", ");
+        Button viewAlterationsButton = convertView.findViewById(R.id.view_alterations_button);
+        viewAlterationsButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent i = new Intent(context,TestAlterationsActivity.class);
+                TestAlterationsActivity.entry = entries[position];
+                context.startActivity(i);
```

Comment:
You are passing to TestAlterationsActivity the value entries[position] as TestElement. Couldn't you just implement into TestResultActivity this part of TestAlterationsActivity ??
```
        // Set the Tags text
        TextView tags = convertView.findViewById(R.id.tags_view);
        StringBuilder assignedTags = new StringBuilder();

        for(String tag: entry.getAlterationTags(entry.getAlterationsNames()[position])) {
            assignedTags.append(tag).append(", ");
        }
        tags.setText(assignedTags.toString());
```
In this way we don't use another activity and adapter.

Reply:
where 
`entry=entries[position];`

Reply:
Oh ok, you would like to show also the score and the extracted text of the alterations

Reply:
Thaks for reviewing.
Your comment has showed me a thing that I've missing.
So yes, I'd like to show the details of the alterations in a new specific activity but I've would like also to show a summary of the alterations test. So I've prepared the string for doing that but I've not used for that.
I'll add it soon.
Thanks.